ArduPlane/servos.cpp

```diff
@@ -464,6 +464,15 @@ void Plane::set_servos_controlled(void)
         // ask quadplane code for forward throttle
         SRV_Channels::set_output_scaled(SRV_Channel::k_throttle, quadplane.forward_throttle_pct());
     }
+    // For ArduSoar
+    
+    if ((control_mode == FLY_BY_WIRE_B ||
+        control_mode == AUTO || control_mode == LOITER) &&
```

Comment:
this should be pushed into AP_Soar: change to .is_active(control_mode) and your system will handle the details of disabling and/or deciding if it is suppressed without cluttering the main plane code. The point is to move as much of the Soaring specific code into the soaring library.